src/engine/GroupByHashMapOptimization.h

```diff
@@ -95,15 +84,11 @@ struct ExtremumAggregationData {
 
   // _____________________________________________________________________________
   [[nodiscard]] ValueId calculateResult(LocalVocab* localVocab) const {
-    auto valueIdResultGetter = [](ValueId id) { return id; };
-    auto stringResultGetter =
-        [localVocab](const ad_utility::triple_component::LiteralOrIri& str) {
-          auto localVocabIndex = localVocab->getIndexAndAddIfNotContained(str);
-          return ValueId::makeFromLocalVocabIndex(localVocabIndex);
-        };
-    return std::visit(ad_utility::OverloadCallOperator(valueIdResultGetter,
-                                                       stringResultGetter),
-                      currentValue_);
+    return std::visit(
+        ad_utility::OverloadCallOperator{
+            [](ValueId id) { return id; },
+            sparqlExpression::detail::makeStringResultGetter(localVocab)},
+        currentValue_);
```

Comment:
1. I think you can move ALL the `calculateResult` functions to the `.cpp`, this includes this function.
1b. Then the header include of the `...Generators.h` is then unused, please remove it.
3. The `visit(Overload{valueIdHandler, makeResultGetter}))`  is also repeateded. Please also move this wrapper to the `SparqlExpressionGenerators.h`.

Reply:
This function is within a templated class, so that's why I didn't move it.